### PR TITLE
CLI: fixing account import ux bugs

### LINKED ISSUE #13162: Wallet directory is asked two times on `./prysm.sh validator accounts import`
### Describe the bug

**Actual behaviour:**

When running the `validator accounts import` command

Exemple:
```
./prysm.sh validator accounts import --keys-dir=~/Downloads/staking_deposit-cli-fdab65d-darwin-amd64/validator_keys --holesky
```

Then the prompt `Enter a wallet directory` is displayed twice:
```
Enter a wallet directory (default: /Users/manu/Library/Eth2Validators/prysm-wallet-v2):
~/Downloads/aaa
Enter a wallet directory (default: /Users/manu/Library/Eth2Validators/prysm-wallet-v2):
~/Downloads/bbb
```
It seems the first prompt has no effect at all. Only the second is used, where the keystore can be found after running the `validator accounts import` command.

**Expected behaviour:**
The prompt `Enter a wallet directory` should be asked (and used) only once.

### Platform(s)

Mac (Apple Silicon)

### What version of Prysm are you running? (Which release)

`v4.1.1`

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

 Bug fix

**What does this PR do? Why is it needed?**

- prevents asking for wallet dir twice when a wallet dir does not exist
- removes keymanager initialization on wallet creation for the local keystore type ( as part of cleanup)
- adds feature for reading sub directories while importing keystores
- fixes a print issue on accounts when imported

example of run:
image

**Which issues(s) does this PR fix?**

Fixes #13162

### CODE REVIEW

**Review by nalepae — APPROVED** [review]

**Comment by nalepae** [comment]
Tested successfully:

```
 bazel run //cmd/validator -- accounts import --keys-dir=<...>
WARNING: Download from https://mirror.bazel.build/github.com/golang/sys/archive/refs/tags/v0.14.0.zip failed: class java.io.FileNotFoundException GET returned 404 Not Found
WARNING: Download from https://mirror.bazel.build/github.com/googleapis/go-genproto/archive/bbf56f31fb17579209f44dda2d2567d20ff21a20.zip failed: class java.io.FileNotFoundException GET returned 404 Not Found
INFO: Analyzed target //cmd/validator:validator (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //cmd/validator:validator up-to-date:
  bazel-bin/cmd/validator/validator_/validator
INFO: Elapsed time: 0.166s, Critical Path: 0.04s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/cmd/validator/validator_/validator accounts import '--keys-dir=<...>'
[2023-12-13 17:03:10]  WARN flags: Running on Ethereum Mainnet
Enter a wallet directory (default: /Users/manu/Library/Eth2Validators/prysm-wallet-v2):

Password requirements: at least 8 characters
New wallet password:
Confirm password:
[2023-12-13 17:03:17]  INFO wallet: Successfully created new wallet wallet-path=/Users/manu/Library/Eth2Validators/prysm-wallet-v2
[2023-12-13 17:03:17]  WARN validator: You are using an insecure gRPC connection. If you are running your beacon node and validator on the same machines, you can ignore this message. If you want to know how to enable secure connections, see: https://docs.prylabs.network/docs/prysm-usage/secure-grpc
[2023-12-13 17:03:17]  INFO accounts: importing validator keystores...
[2023-12-13 17:03:17]  INFO accounts: checking directory for keystores: <...>
Enter the password for your imported accounts:
Importing accounts, this may take a while...
Importing accounts... 100% [==========================================]  [0s:0s]
[2023-12-13 17:03:20]  INFO local-keymanager: Successfully imported validator key(s) publicKeys=0xa69a5c7a0c8b
[2023-12-13 17:03:20]  INFO accounts: Imported accounts [a69a5c7a0c8b20f78a09b0837bc96463e19f4df2436192b44e088c50f339ef162d8c6288b50a533b9c5b781fd1726b72], view all of them by running `accounts list`
```

File: validator/accounts/wallet/wallet.go [hunk 1]
```diff
@@ -193,6 +194,65 @@ func OpenWalletOrElseCli(cliCtx *cli.Context, otherwise func(cliCtx *cli.Context
 	})
 }
 
+// OpenOrCreateNewWallet takes a cli and returns a wallet either opening an existing valid wallet or creating a new one.
+func OpenOrCreateNewWallet(cliCtx *cli.Context) (*Wallet, error) {
+	walletDir, err := accountsprompt.InputDirectory(cliCtx, accountsprompt.WalletDirPromptText, flags.WalletDirFlag)
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (james-prysm) [reply validator/accounts/wallet/wallet.go:201]: This is part of a greater refactoring, mike neuder was first to help on this but I think there is more to look into.
there are a few other places that do checks and it's around `waiting for activation` which is specific to the use of web ui in some other places we also don't give the user experience to try again ( it just fails the first time) but it seems appropriate on import.

[the remark this answers is not in the document]
Comment (james-prysm) [reply validator/accounts/wallet/wallet.go:201]: https://github.com/prysmaticlabs/prysm/issues/9883 issue is closed but there are still remnants of it i suppose

File: validator/accounts/wallet/wallet.go [hunk 2]
```diff
@@ -193,6 +194,65 @@ func OpenWalletOrElseCli(cliCtx *cli.Context, otherwise func(cliCtx *cli.Context
...
+		if err != nil {
+			return nil, err
+		}
+		return OpenWallet(cliCtx.Context, &Config{
+			WalletDir:      walletDir,
+			WalletPassword: walletPassword,
+		})
+	}
+	// create a new wallet in the dir
+	walletPassword, err := prompt.InputPassword(
```
[the remark this answers is not in the document]
Comment (james-prysm) [reply validator/accounts/wallet/wallet.go:231]: was not intentional, spent time fixing test to address this.

**Review by nalepae — APPROVED** [review]

**Pull request merged**